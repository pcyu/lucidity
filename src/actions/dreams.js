import uuid from 'uuid'

// ADD_DREAM
export const addDream = ({ title = '', entry = '', hoursSlept = 0, dreamType = '', createdAt = 0, lucidityLevel = 0 } = {}) => ({
  type: 'ADD_DREAM',
  dream: {
    id: uuid(),
    title,
    entry,
    hoursSlept,
    dreamType,
    createdAt,
    lucidityLevel
  }
})

// REMOVE_DREAM
export const removeDream = ({ id }) => ({
  type: 'REMOVE_DREAM',
  id
})

// EDIT_DREAM
export const editDream = (id, updates) => ({
  type: 'EDIT_DREAM',
  id,
  updates
})