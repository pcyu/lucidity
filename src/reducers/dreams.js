// Dreams Reducer
const dreamsReducerDefaultState = []

const dreamsReducer = (state = dreamsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_DREAM':
      return [
        ...state,
        action.dream
      ]
    case 'REMOVE_DREAM':
      return state.filter(({ id }) => id !== action.id)
    case 'EDIT_DREAM':
      return state.map((dream) => {
        if (dream.id === action.id) {
          return {
            ...dream,
            ...action.updates
          }
        } else {
          return dream
        }
      })
    default:
      return state
  }
}

export default dreamsReducer