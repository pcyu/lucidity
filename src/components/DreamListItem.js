import React from 'react'
import { removeDream } from '../actions/dreams'
import { Link } from 'react-router-dom'
import moment from 'moment'

class DreamListItem extends React.Component {
  constructor(props) {
    super(props)
    this.openDream = this.openDream.bind(this)
  }
  openDream() {
    const dreamBody = this.refs.dreamBody
    dreamBody.style.maxHeight ? 
      dreamBody.style.maxHeight = null : dreamBody.style.maxHeight = dreamBody.scrollHeight + 'px'
  }
  render() {
    return (
      <div className="dream">
        <div className="dream__header" onClick={this.openDream}>
          <span className="dream__title">{this.props.title}</span>
          <span className="dream__type">{this.props.dreamType}</span>
          <span className="dream__hours">{this.props.hoursSlept}</span>
          <span className="dream__date">{moment(this.props.createdAt).format('MMM Do, YYYY')}</span>
        </div>
        <div className="dream__body" ref="dreamBody">
          <p className="dream__entry"><i class="fas fa-quote-left dream__quote"></i>{this.props.entry}</p>
        </div>

            {/*<Link to={`/edit/${id}`} className="dream__link">
              </Link>*/}
      </div>
    )
  }
}

// const DreamListItem = ({ id, title, entry, dreamType, createdAt, hoursSlept }) => (
//   <div className="dream">
//     <div className="dream__header">
//       <span className="dream__title">{title}</span>
//       <span className="dream__type">{dreamType}</span>
//       <span className="dream__hours">{hoursSlept}</span>
//       <span className="dream__date">{moment(createdAt).format('MMM Do, YYYY')}</span>
//     </div>
//     <div className="dream__body">
//       <p className="dream__entry">{entry}</p>
//     </div>

//     {/*<Link to={`/edit/${id}`} className="dream__link">
// </Link>*/}
//   </div>
// )

export default DreamListItem