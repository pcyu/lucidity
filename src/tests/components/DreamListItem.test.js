import React from 'react'
import { shallow } from 'enzyme'
import DreamListItem from '../../components/DreamListItem'
import dreams from '../fixtures/dreams'

test('should render Dream list item', () => {
  const wrapper = shallow(<DreamListItem {...dreams[0]} />)
  expect(wrapper).toMatchSnapshot()
})