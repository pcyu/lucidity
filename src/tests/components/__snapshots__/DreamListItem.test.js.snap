// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dream list item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      title: 
      dream one
    </h3>
  </Link>
  <p>
    entry: 
    this is dream one entry
  </p>
  <p>
    dreamType: 
    normal
  </p>
  <p>
    createdAt: 
    0
  </p>
  <p>
    hours slept: 
    6
  </p>
</div>
`;
